lib/extensions/category_extension.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module DiscourseChat::CategoryExtension
+  extend ActiveSupport::Concern
+
+  prepended do
+    has_one :chat_channel, as: :chatable
+  end
+
+  def cannot_delete_reason
+    return I18n.t('category.cannot_delete.has_chat_channels') if chat_channel
```

Comment:
If there is more than one reason, shouldn't we combine them?

Reply:
The original method is this one:
```ruby
  def cannot_delete_reason
    return I18n.t('category.cannot_delete.uncategorized') if self.uncategorized?
    return I18n.t('category.cannot_delete.has_subcategories') if self.has_children?

    if self.topic_count != 0
      oldest_topic = self.topics.where.not(id: self.topic_id).order('created_at ASC').limit(1).first
      if oldest_topic
        I18n.t('category.cannot_delete.topic_exists', count: self.topic_count, topic_link: "<a href=\"#{oldest_topic.url}\">#{CGI.escapeHTML(oldest_topic.title)}</a>")
      else
        # This is a weird case, probably indicating a bug.
        I18n.t('category.cannot_delete.topic_exists_no_oldest', count: self.topic_count)
      end
    end
  end
```
So I guess we could combine messages but the original implementation doesn’t seem to care about this (a category can have subcategories _and_ topics). Not sure it’s worth the time?

Reply:
I don't think so, no. Thanks for clarifying!